WeatherMonitoringSystem/Domain/Config/ConfigurationManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿
+using WeatherMonitoringSystem.Config;
+
+namespace WeatherMonitoringSystem.Domain.Config
+{
+    using Newtonsoft.Json;
+    public class ConfigurationManager
+    {
+        private static ConfigurationManager _instance;
+        private static readonly object _lock = new object();
+
+        public BotConfig RainBot { get; private set; }
+        public BotConfig SunBot { get; private set; }
+        public BotConfig SnowBot { get; private set; }
+
+        private ConfigurationManager()
+        {
+                LoadConfiguration();
+        }
+
+        public static ConfigurationManager Instance
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _instance ??= new ConfigurationManager();
+                }
+            }
+        }
+
+        private void LoadConfiguration()
+        {
+            try
+            {
+                string relativePath = @"JSON_File\botConfig.json";
+                string configPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, relativePath);
+                if (!File.Exists(configPath))
+                {
+                    throw new FileNotFoundException("Configuration file not found.");
+                }
+
+                string json = File.ReadAllText(configPath);
+                dynamic config = JsonConvert.DeserializeObject<dynamic>(json);
```

Comment:
We usually prefer strongly typed models not dynamic types. Lets try 
 'JsonConvert.DeserializeObject<BotsConfiguration>(json)', and configure a new model for the config model `BotsConfiguration` to have three bots configuration